### ISSUE TITLE
GTAIV.EFLC.FusionFix v3.5.1 Cutscene audio delay

### ISSUE THREAD

**Issue by chunkybubble** [ThirteenAG/GTAIV.EFLC.FusionFix#691]
### What should happen

Cutscene audio plays and is synced with character mouth movements and timing.
I used the latest version v3.5.1 and the cutscene was 3-5 seconds ahead of the cutscene audio.

Tried v3.4.0 and it worked as it should.

### Checklist

- [X] Check the [latest git build](https://ci.appveyor.com/api/projects/ThirteenAG/gtaiv-eflc-fusionfix/artifacts/GTAIV.EFLC.FusionFix.zip) in case it's already implemented.
- [X] [Search for other requests](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/search?q=my+issue&type=issues) of the same feature.

**Comment by ThirteenAG** [comment]
https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/blob/bc3fc1c1a63ecf46c79d04c606f98b401a9d05e3/data/plugins/GTAIV.EFLC.FusionFix.ini#L46

You can disable the latest cutscene tweak with this option. Not everyone have that issue, so we will have to wait and see what to do about it. Try capping cutscene fps to 30 or 60, maybe it will be better that way.

**Comment by Yujah** [comment]
FWIW another thread on Steam about this:

https://steamcommunity.com/app/12210/discussions/0/664957578230381716/

Using audio over DP, as I would imagine might be the thing there (DP or HDMI, that is). ExtraCutSceneFix=0 fixed it for them.

No idea about the numbers, but it's the first such thread I noticed. Sort of unfortunate if it's common: have been spamming FusionFix advise into the Steam forum basically daily.

**Comment by ThirteenAG** [comment]
Well, what else can I do? I can add a menu toggle for more convenient access to that toggle, but disabling it will return cutscene jitter. Obviously the solution is not ideal, but it somehow works [more often than not?]

**Comment by ThirteenAG** [comment]
If anybody can reproduce the delay consistently and willing to test a few builds, let me know.

**Comment by Yujah** [comment]
Yes, a menu-toggle would probably be good, if this is not just rare -- and maybe if it's very non-rare, defaulting it to off. But...

I tried using DP-audio on my own setup (GTX970 to an Iiyama monitor) and I can't myself reproduce. Have also as said not seen the issue before on the Steam-forum so I'd assume it's not common at least

**Comment by Yujah** [comment]
Update: reporter got back to the issue; appears it's not this ExtraCutSceneFix thing after all:

https://steamcommunity.com/app/12210/discussions/0/664957578230381716/?tscn=1750367620#c591774577804222495

The multi-second delay seemed to make it likely that it was; never saw that reported in another context than this one here, but I guess this may want to be closed again then.

**Comment by ClaudeLib** [comment]
Out of curiosity, I lowered the graphics settings and resolution, giving the game some performance headroom, and when running at ~300 fps, the cutscenes played almost comically fast, going out of sync very fast. But at this point the gameplay also became too fast. My point is, a) maybe these are related, and b) the gameplay speedup was claimed to be fixed in the other patch, so maybe this is something worth looking into. The test was performed in TBoGT, running on SteamOS.
EDIT: Weirdly enough, the base IV intro cutscene doesn't seem to have the audio desync issue, even when running at 144 fps, judging by the moment where Niko closes the trunk of Roman's Cab at the end of the cutscene, but in TBoGT, Tony's doorbell buzz is clearly out of sync even at 60 fps. I'll test this further.
EDIT2: I didn't know this issue was open when creating #756, my bad.

**Comment by Yujah** [comment]
Yes, the game's general high-FPS issues are very well known. See e.g.

https://github.com/GTAmodding/GTAIV-Issues-List/issues/112

I in the Steam forum for most any issue first of all advise explicitly frame-capping to under 60 FPS either with an external or refresh-rate+vsync cap or FusionFix' FPS-limiter before feeling it worth to even look closer.

If you went to 300 FPS by just lowering settings you are with not lowered settings still above 60, and will see all of the issues from that list (as it shows, 30 FPS is already a boundary but above 60 things start becoming impossible to ignore).

**Comment by ClaudeLib** [comment]
With the latest build the desync issue appears to be mostly mitigated, but not eliminated completely. Weirdly enough, when playing the base IV intro cutscene, it seems to go in perfect sync, but in TBoGT, the animation is 0.3-0.4 seconds ahead of audio by the end of the cutscene, evidenced by the noticeable jump when enabling the cutscene audio sync by pressing Up.
Testing environment: AMD PC running vanilla SteamOS, connected to a display via DisplayPort at 1080p60.
Will do further testing on OLEDeck later.

**Comment by ClaudeLib** [comment]
Tested the latest build on OLEDeck at 90 fps.
It appears the TBoGT intro cutscene already goes out of sync at the very beginning, roughly 13 frames in the 30 fps recording. After manually resyncing and playing through the whole cutscene at somewhat consistent 90 fps the desync in the end is very miniscule, 6 frames.
https://streamable.com/xjd8fn

**Comment by ClaudeLib** [comment]
For some reason I couldn't record video on my PC running SteamOS, but I had to switch to Windows temporarily for unrelated reason, so I gave it a try again. Here's the game running at unstable 144 fps on Windows 11: https://streamable.com/7i9fb6
The desync issue is much more severe. Weirdly enough, when testing on SteamOS, the desync appeared to be the same as in the 90 fps test. I couldn't count the exact number of frames, because the recording kept getting corrupted, but it certainly appeared bearable. I am not sure if this has anything to do with DXVK, because when trying to run the game with it, it just kept freezing after about a minute, so I ran the game on pure DX9.

**Comment by ClaudeLib** [comment]
This is so inconsistent. I ran the game again, testing something else, still on Windows, but now the desync was again barely noticeable. Same FF build, same system, even same session. Here's the recording I made at the very end of the cutscene: https://streamable.com/gfa0eb
